pkg/audit/containment_test.go

```diff
@@ -0,0 +1,127 @@
+			NodeID:            planet.StorageNodes[0].ID(),
+			PieceID:           storj.PieceID{},
+			StripeIndex:       0,
+			ShareSize:         0,
+			ExpectedShareHash: someHash,
+			ReverifyCount:     0,
+		}
+
+		err = planet.Satellites[0].DB.Containment().IncrementPending(ctx, input)
+		require.NoError(t, err)
+
+		output, err := planet.Satellites[0].DB.Containment().Get(ctx, input.NodeID)
+		require.NoError(t, err)
+
+		require.Equal(t, input, output)
+	})
+}
+
+func TestContainIncrementPendingEntryExists(t *testing.T) {
+	testplanet.Run(t, testplanet.Config{
+		SatelliteCount: 1, StorageNodeCount: 4,
+	}, func(t *testing.T, ctx *testcontext.Context, planet *testplanet.Planet) {
+
+		randBytes := make([]byte, 10)
+		_, err := rand.Read(randBytes)
+		require.NoError(t, err)
+		hash1 := pkcrypto.SHA256Hash(randBytes)
+
+		info1 := &audit.PendingAudit{
+			NodeID:            planet.StorageNodes[0].ID(),
+			PieceID:           storj.PieceID{},
+			StripeIndex:       0,
+			ShareSize:         0,
+			ExpectedShareHash: hash1,
+			ReverifyCount:     0,
+		}
+
+		err = planet.Satellites[0].DB.Containment().IncrementPending(ctx, info1)
+		require.NoError(t, err)
+
+		randBytes = make([]byte, 10)
+		_, err = rand.Read(randBytes)
+		require.NoError(t, err)
+		hash2 := pkcrypto.SHA256Hash(randBytes)
+
+		info2 := &audit.PendingAudit{
+			NodeID:            planet.StorageNodes[0].ID(),
+			PieceID:           storj.PieceID{},
+			StripeIndex:       1,
+			ShareSize:         1,
+			ExpectedShareHash: hash2,
+			ReverifyCount:     0,
+		}
+
+		err = planet.Satellites[0].DB.Containment().IncrementPending(ctx, info2)
+		require.NoError(t, err)
+
+		pending, err := planet.Satellites[0].DB.Containment().Get(ctx, info1.NodeID)
+		require.NoError(t, err)
+
+		require.Equal(t, 1, pending.ReverifyCount)
+
+		// Even though info2 had different pending audit info,
+		// we only need to save the original pending audit info
+		// since the node ID is the same.
+		require.Equal(t, info1.ExpectedShareHash, pending.ExpectedShareHash)
+	})
+}
+
+func TestContainDelete(t *testing.T) {
+	testplanet.Run(t, testplanet.Config{
+		SatelliteCount: 1, StorageNodeCount: 4,
+	}, func(t *testing.T, ctx *testcontext.Context, planet *testplanet.Planet) {
+
+		randBytes := make([]byte, 10)
+		_, err := rand.Read(randBytes)
+		require.NoError(t, err)
+		hash1 := pkcrypto.SHA256Hash(randBytes)
+
+		info1 := &audit.PendingAudit{
+			NodeID:            planet.StorageNodes[0].ID(),
+			PieceID:           storj.PieceID{},
+			StripeIndex:       0,
+			ShareSize:         0,
+			ExpectedShareHash: hash1,
+			ReverifyCount:     0,
+		}
+
+		err = planet.Satellites[0].DB.Containment().IncrementPending(ctx, info1)
+		require.NoError(t, err)
+
+		err = planet.Satellites[0].DB.Containment().Delete(ctx, info1.NodeID)
+		require.NoError(t, err)
+
+		_, err = planet.Satellites[0].DB.Containment().Get(ctx, info1.NodeID)
+		require.Error(t, err, audit.ErrContainedNotFound.New(info1.NodeID.String()))
```

Comment:
Shouldn't we verify that the returned error is of such specific class?

Reply:
why would that be better than as it is currently?

Reply:
I didn't mean to remove this one, I was asking for having one `require` more which checks that the error is of `ErrContainedNotFound`.

Currently it doesn't do such check, your are writing the message to print out if the assertion fails, see docs https://godoc.org/github.com/stretchr/testify/require#Error

Reply:
okay thanks, I went ahead and added the extra check!